### ISSUE TITLE
[data grid] The provided aria label prop is attached on the wrong grid element, causing issues with accessibility

### ISSUE THREAD

**Issue by lemmylem** [mui/mui-x#13522]
### The problem in depth

I am a MUI Pro license user and customers are experiencing accessibility issues with the data grid. The problem is that the aria label that MUI provides is attached to the "wrong" element. It should be attached to the div with the `role="grid"`, so screen readers can pick it up.

Currently, for example, the NVDA screen reader does not correctly read the aria label, in the grid, failing our accessibility requirements. In the below image, the red circle is where the aria label is currently attached, and the green is where it should be (or vise versa). The overall requirement is that the role must be on the same div as the aria label, or else there will be issues with screen readers reading the correct back to the user.

<img width="1323" alt="aria-label" src="https://github.com/mui/mui-x/assets/139143259/6df5d4a7-a844-46de-b554-5e9b93d320dd">

**Search keywords**: aria, label
**Order ID**: 79357

**Comment by michelengelen** [comment]
Hey @lemmylem and thanks for raising this issue.
I can confirm that the aria values are incorrectly placed.
I'll add it to the board! 👍🏼

**Comment by samwato** [comment]
I also ran into this issue using the JAWS screen reader. It also requires the aria label on the same div with `role="grid"`.

I ended up quickly patching `x-data-grid` to get by for now.

```
diff --git a/node_modules/@mui/x-data-grid/hooks/utils/useGridAriaAttributes.js b/node_modules/@mui/x-data-grid/hooks/utils/useGridAriaAttributes.js
index 238b409..354e014 100644
--- a/node_modules/@mui/x-data-grid/hooks/utils/useGridAriaAttributes.js
+++ b/node_modules/@mui/x-data-grid/hooks/utils/useGridAriaAttributes.js
@@ -20,6 +20,8 @@ export const useGridAriaAttributes = () => {
     role,
     'aria-colcount': visibleColumns.length,
     'aria-rowcount': headerGroupingMaxDepth + 1 + pinnedRowsCount + totalRowCount,
-    'aria-multiselectable': isMultipleRowSelectionEnabled(rootProps)
+    'aria-multiselectable': isMultipleRowSelectionEnabled(rootProps),
+    'aria-busy': rootProps.forwardedProps['aria-busy'],
+    'aria-label': rootProps.forwardedProps['aria-label']
   };
 };
```

Other's may also need the `aria-describedby` which seems to have gone as well.

**Comment by lemmylem** [comment]
> Hey @lemmylem and thanks for raising this issue. I can confirm that the aria values are incorrectly placed. I'll add it to the board! 👍🏼

Do you guys have an issue tracker board that tells users what you guys are working on or when certain bugs will be released? @michelengelen

**Comment by samwato** [comment]
Any updates on the status of this issue?
I'm happy to submit a PR as a starting point.

**Comment by michelengelen** [comment]
> Any updates on the status of this issue?
> 
> I'm happy to submit a PR as a starting point.

That would be awesome! Feel free to tag me on that PR so I can assign the right people to it! 💪🏼
